### PR TITLE
migrate set generation to go_genrule

### PR DESCRIPTION
**Pull request by mikedanese**
Depends on https://github.com/kubernetes/release/pull/238

### CODE REVIEW

**Comment by spxtr** [comment]
Please reassign when this is ready for review.

**Comment by spxtr** [comment]
@k8s-bot bazel test this
@k8s-bot verify test this

**Comment by spxtr** [comment]
```
I0501 10:43:52.365] FAILED   hack/make-rules/../../hack/verify-golint.sh	39s
I0501 10:43:52.366] Skipping hack/make-rules/../../hack/verify-govet.sh
I0501 10:43:52.366] Verifying hack/make-rules/../../hack/verify-import-boss.sh
W0501 10:43:52.466] Some packages passed golint but are not listed in hack/.linted_packages.
W0501 10:43:52.466] Please add them in alphabetical order:
W0501 10:43:52.466] 
W0501 10:43:52.467]   echo pkg/util/sets/types >> hack/.linted_packages
W0501 10:43:52.467]   LC_ALL=C sort -o hack/.linted_packages hack/.linted_packages
W0501 10:43:52.467] 
W0501 10:43:52.467] You can test via this script and commit the result.
```

**Comment by spxtr** [comment]
@k8s-bot unit test this
#45170

**Comment by spxtr** [comment]
OK
/lgtm

File: staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD [hunk 1]
```diff
@@ -2,28 +2,57 @@ package(default_visibility = ["//visibility:public"])
...
 )
 
-go_test(
-    name = "go_default_test",
-    srcs = ["set_test.go"],
-    library = ":go_default_library",
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "byte.go",
```
Comment (spxtr) [inline staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD]: Can we delete these files?

Comment (mikedanese) [inline staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD]: Not until we don't use go build.

**Comment by mikedanese** [comment]
@k8s-bot unit test this

**Comment by mikedanese** [comment]
@thockin can you look at this? needs approval.

**Pull request merged**